hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/basic/BasicCriteriaUsageTest.java

```diff
@@ -164,4 +145,106 @@ public void testDoubleNegation() {
 		em.getTransaction().commit();
 		em.close();
 	}
+
+	@Entity( name = "Payment" )
+	@Table( name = "crit_basic_payment" )
+	public static class Payment {
+
+		private Long id;
+		private BigDecimal amount;
+		private Date date;
+
+		@Id
+		@GeneratedValue
+		public Long getId() {
+			return id;
+		}
+
+		public void setId(Long id) {
+			this.id = id;
+		}
+
+		public BigDecimal getAmount() {
+			return amount;
+		}
+
+		public void setAmount(BigDecimal amount) {
+			this.amount = amount;
+		}
+
+		@Column( name = "payment_date" )
+		public Date getDate() {
+			return date;
+		}
+
+		public void setDate(Date date) {
+			this.date = date;
+		}
+	}
+
+	@Entity( name = "Wall" )
+	@Table( name = "crit_basic_wall" )
+	public static class Wall {
+		private Long id;
+		private long width;
+		private long height;
+		private String color;
+		private Wall left;
+		private Wall right;
+
+		@Id
+		@GeneratedValue
+		public Long getId() {
+			return id;
+		}
+
+		public void setId(Long id) {
+			this.id = id;
+		}
+
+		public long getWidth() {
+			return width;
+		}
+
+		public void setWidth(long width) {
+			this.width = width;
+		}
+
+		public long getHeight() {
+			return height;
+		}
+
+		public void setHeight(long height) {
+			this.height = height;
+		}
+
+		public String getColor() {
+			return color;
+		}
+
+		public void setColor(String color) {
+			this.color = color;
+		}
+
+		@ManyToOne
+		@JoinColumn(name = "left_id")
+		public Wall getLeft() {
+			return left;
+		}
+
+		public void setLeft(Wall left) {
+			this.left = left;
+		}
+
+		@ManyToOne
+		@JoinColumn(name = "right_id")
+		public Wall getRight() {
+			return right;
+		}
+
+		public void setRight(Wall right) {
+			this.right = right;
+		}
+	}
```

Comment:
Why did you inline the entity classes here and not just reuse the previous ones?

Reply:
My original thought is it is only used within BasicCriteriaUsageTest and making them inline improves code readability (the testing case doesn't require apt result so inline is acceptable). Now I revert back the changes to make testing code comparison easier.